File: AtCoder/ACP/BootCamp4b/Medium100_2/python/A_Airport_Bus.py
N, C, K = map(int, input().split())
T = sorted([int(input()) for _ in range(N)])

t = 0
cnt = 0
res = 0
for i in range(N):
    if T[i] > t:
        res += 1
        cnt = 0
        t = T[i] + K
    cnt += 1
    if cnt > C:
        res += 1
        cnt = 1
        t = T[i] + K
print(res)

File: AtCoder/ACP/BootCamp4b/Medium100_2/cpp/A_Airport_Bus.cpp
#include <bits/stdc++.h>
#define MAX_N 100'000
using namespace std;
typedef long long ll;

ll N, C, K;
ll T[MAX_N];

void solve() {
  sort(T, T + N);
  ll res = 0;
  ll cnt = 0;
  ll t = 0;
  for (int i = 0; i < N; ++i) {
    if (T[i] > t) {
      ++res;
      cnt = 0;
      t = T[i] + K;
    }
    ++cnt;
    if (cnt > C) {
      ++res;
      cnt = 1;
      t = T[i] + K;
    }
  }
  printf("%lld\n", res);
}

int main() {
  cin >> N >> C >> K;
  for (int i = 0; i < N; ++i) {
    cin >> T[i];
  }
  solve();
}